hedera-mirror-importer/src/main/resources/db/migration/V1.31.0__remove_invalid_entities.sql

```diff
@@ -0,0 +1,10 @@
+-------------------
+-- Fix entity type mismatches
+-------------------
+
+-- Backup t_entities prior to type mismatch fix
+create table t_entities_archive as table t_entities;
+
+--- Delete entities above the max entity id witnessed by transactions
+delete from t_entities where id > (select max(entity_id) from transaction where entity_id is not null);
```

Comment:
I'm not convinced this is correct. I believe we should be deleting anything above the max for only successful transaction results. Below results for mainnet indicate everything above 58743 should be deleted:
```
select max(entity_id) as max, result from transaction where entity_id is not null group by result order by result asc;
  max   | result
--------+--------
  45009 |      4
  56708 |      7
  51273 |     10
  45009 |     11
  40319 |     15
  58743 |     22
  24586 |     30
  54356 |     33
  22922 |     34
  55762 |     68
  35316 |     72
  40790 |     78
    112 |    105
  15602 |    106
  40941 |    111
 211228 |    150
  40127 |    158
(17 rows)
```

Reply:
Yep, good catch. Should be 
```suggestion
delete from t_entities where id > (select max(entity_id) from transaction where entity_id is not null and result = 22);
```

Reply:
Updated